store/task.go

```diff
@@ -131,6 +131,19 @@ func (s *Store) FindTask(ctx context.Context, find *api.TaskFind, returnOnErr bo
 		}
 		taskList = append(taskList, task)
 	}
+
+	// Filter tasks belongs to archived instances.
+	{
+		var filteredList []*api.Task
```

Comment:
We may want to preserve tasks because tasks are similar to records of issues/pipelines. We want to preserve it even if the instances are archived.